celest/schedule/removal_operators.py

```diff
@@ -0,0 +1,68 @@
+
+
+import random
+
+
+def random_removal(request_handler, number_to_remove):
+
+    if number_to_remove > request_handler.number_of_scheduled_requests:
+        request_handler.unschedule_all_requests()
+        return request_handler
+
+    while (number_to_remove > 0):
+        request_index = random.randint(0, len(request_handler.requests) - 1)
+        if request_handler.is_request_scheduled(request_index):
+            request_handler.unschedule_request(request_index)
+            number_to_remove -= 1
+
+    return request_handler
+
+
+def priority_removal(request_handler, number_to_remove):
+
+    if number_to_remove >= request_handler.number_of_scheduled_requests:
+        request_handler.unschedule_all_requests()
+        return request_handler
+
+    request_handler.sort_by_decreasing_priority()
+    return remove_first_n_scheduled_requests(request_handler, number_to_remove)
+
+
+def remove_first_n_scheduled_requests(request_handler, number_to_remove):
+
+    index = 0
+    while (number_to_remove > 0):
```

Comment:
Stop the while loop if the index goes beyond the number of requests. Currently, it doesn't look like there anything that would stop the iteration if the number of scheduled requests is less then the number to remove.

Reply:
There are checks in the removal functions that call `remove_first_n_scheduled_requests` to ensure that there are less items to remove then scheduled requests. I have no incorporated these checks into the `remove_first_n_scheduled_requests` function.